.github/workflows/ci.yml

```diff
@@ -0,0 +1,69 @@
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        requirements:
+          [
+            { php-version: '7.1', symfony-version: '3.4.*', dependency-versions: 'lowest' },
+            { php-version: '7.1', symfony-version: '3.4.*', dependency-versions: 'latest' },
+            { php-version: '7.2', symfony-version: '3.4.*', dependency-versions: 'lowest' },
+            { php-version: '7.2', symfony-version: '3.4.*', dependency-versions: 'latest' },
+            { php-version: '7.3', symfony-version: '3.4.*', dependency-versions: 'lowest' },
+            { php-version: '7.3', symfony-version: '3.4.*', dependency-versions: 'latest' },
+            { php-version: '7.4', symfony-version: '3.4.*', dependency-versions: 'lowest' },
+            { php-version: '7.4', symfony-version: '3.4.*', dependency-versions: 'latest' },
+            { php-version: '7.1', symfony-version: '4.4.*', dependency-versions: 'lowest' },
+            { php-version: '7.1', symfony-version: '4.4.*', dependency-versions: 'latest' },
+            { php-version: '7.2', symfony-version: '4.4.*', dependency-versions: 'lowest' },
+            { php-version: '7.2', symfony-version: '4.4.*', dependency-versions: 'latest' },
+            { php-version: '7.3', symfony-version: '4.4.*', dependency-versions: 'lowest' },
+            { php-version: '7.3', symfony-version: '4.4.*', dependency-versions: 'latest' },
+            { php-version: '7.4', symfony-version: '4.4.*', dependency-versions: 'lowest' },
+            { php-version: '7.4', symfony-version: '4.4.*', dependency-versions: 'latest' },
+#            { php-version: '8.0', symfony-version: '4.4.*', dependency-versions: 'lowest' },
+            { php-version: '8.0', symfony-version: '4.4.*', dependency-versions: 'latest' },
+#            { php-version: '8.1', symfony-version: '4.4.*', dependency-versions: 'lowest' },
+            { php-version: '8.1', symfony-version: '4.4.*', dependency-versions: 'latest' },
+#            { php-version: '7.2', symfony-version: '5.4.*', dependency-versions: 'lowest' },
+#            { php-version: '7.2', symfony-version: '5.4.*', dependency-versions: 'latest' },
+#            { php-version: '7.3', symfony-version: '5.4.*', dependency-versions: 'lowest' },
+#            { php-version: '7.3', symfony-version: '5.4.*', dependency-versions: 'latest' },
+#            { php-version: '7.4', symfony-version: '5.4.*', dependency-versions: 'lowest' },
+#            { php-version: '7.4', symfony-version: '5.4.*', dependency-versions: 'latest' },
+#            { php-version: '8.0', symfony-version: '5.4.*', dependency-versions: 'lowest' },
+#            { php-version: '8.0', symfony-version: '5.4.*', dependency-versions: 'latest' },
+#            { php-version: '8.1', symfony-version: '5.4.*', dependency-versions: 'lowest' },
+#            { php-version: '8.1', symfony-version: '5.4.*', dependency-versions: 'latest' },
+          ]
```

Comment:
Let's take another approach. Let's test only the lowest bound of our dependencies with `--prefer-lowest` (in our case PHP 7.1, Symfony 3.x) and all of the others with their highest dependencies. Also, let's only test major versions of Symfony, e.g. `3.*`, `4.*`, etc.


Let's consider an alternative approach. We'll test only the lowest bound of our dependencies using `--prefer-lowest` (in our case, PHP 7.1 and Symfony 3.x), while testing all others with their highest dependencies. Additionally, let's focus on testing major versions of Symfony, such as `3.*`, `4.*`, and so on. And finally, if some combination is not possible, because Symfony does not support some of our PHP versions, let's exclude them using [`jobs.<job_id>.strategy.matrix.exclude`](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#excluding-matrix-configurations).

For example,
```yml
  phpunit:
    name: Run tests PHP ${{ matrix.php }} Symfony ${{ matrix.symfony }} ${{ matrix.dependency }}
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        php:
          - '7.1'
          - '7.2'
          - '7.3'
          - '7.4'
          - '8.0'
          - '8.1'
          - '8.2'
        symfony:
          - '3.*'
          - '4.*'
          - '5.*'
          - '6.*'
        dependency:
            - 'highest'
        include:
          - { php: '7.1', symfony: '3.*', dependency: 'lowest' }
        exclude:
          - { php: '7.2', symfony: '6.*' }
          - { php: '7.2', symfony: '6.*' }
          - { php: '7.3', symfony: '6.*' }
          - { php: '7.3', symfony: '6.*' }
          - { php: '7.4', symfony: '6.*' }
          - { php: '7.4', symfony: '6.*' }
          - { php: '8.0', symfony: '6.*' }
```